### DISCUSSION TITLE
Facebook Login doesn't provide JWTs

### DISCUSSION THREAD

**Discussion by jrch2k20** [abacritt/angularx-social-login#406] [Q&A] [answered]
Hi abacritt.

First tyvm for your work, is a real good component.

Ok, my issue is i'm trying to add social login from your component and i need to use JWT for server side validation, so far Google and Microsoft are working perfectly but Facebook doesn't return idToken like Google or Microsoft do, instead it seems you get the payload directly, console log show this(not empty just redacted)

```
authToken: "EAAP7A6IDZCHQBAPDUT62MnTikNv8bD6uPO8z4CZAeDOsCCv1ReRrQ1TOeqACR8GxOqmxyDZBjcukCECIeZCcRmuqdARHsxO5wuubJW5W4nvaxAY3afTxCYUYZAclTscZBbewrcR0t0DMfbJkz65ZBnjZC34IBMKFQKsINBfLo8LkLnZBQ2rfEoZAdSfyrlgSYQKiqoXuW1PP4trSjGsmqbAZCRU"
​
email: ""
​
firstName: ""
​
id: ""
​
lastName: ""
​
name: ""
​
photoUrl: ""
​
provider: "FACEBOOK"
```

Is there a hidden parameter or anything to inform the component to get the actual JWTs?

Thank you very much for your time.

P.S: Angular 11 with component version 3.5.7

**Comment by jaibatrik** [comment, marked as answer]
Hey, you need to check Facebook's documentation to see how you can get the JWT. I don't recollect any special params or anything for Facebook.
